datasets/irc_disentangle/irc_disentangle.py

```diff
@@ -0,0 +1,270 @@
+
+    def _generate_examples(self, filepath, split):
+        """ Yields examples. """
+
+        if self.config.name == "ubuntu":
+            # run loop for each date
+            all_files = glob.glob(f"{filepath}/*.annotation.txt")
+            all_dates = [file.split("/")[-1][:10] for file in all_files]
+            all_info = [file.split("/")[-1][10:-15] for file in all_files]
+
+        elif self.config.name == "channel_two":
+            # run loop once (no dates for this config)
+            all_dates = ["_"]
+            all_info = ["_"]
+
+        last_id = 0
+        id_ = 0
+
+        for date, info in zip(all_dates, all_info):
+
+            if self.config.name == "ubuntu":
+                # load file of given date and additional info
+                raw_path = os.path.join(filepath, f"{date}{info}.raw.txt")
+                ascii_path = os.path.join(filepath, f"{date}{info}.ascii.txt")
+                tok_path = os.path.join(filepath, f"{date}{info}.tok.txt")
+                annot_path = os.path.join(filepath, f"{date}{info}.annotation.txt")
+
+            elif self.config.name == "channel_two":
+                raw_path = os.path.join(filepath, f"channel-two.{split}.raw.txt")
+                ascii_path = os.path.join(filepath, f"channel-two.{split}.ascii.txt")
+                tok_path = os.path.join(filepath, f"channel-two.{split}.tok.txt")
+                annot_path = os.path.join(filepath, f"channel-two.{split}.annotation.txt")
+
+            with open(raw_path, encoding="utf-8") as f_raw, open(ascii_path, encoding="utf-8") as f_ascii, open(
+                tok_path, encoding="utf-8"
+            ) as f_tok, open(annot_path, encoding="utf-8") as f_annot:
+
+                raw_sentences = f_raw.read().split("\n")
+                ascii_sentences = f_ascii.read().split("\n")
+                tok_sentences = f_tok.read().split("\n")
+                annot_sentences = f_annot.read().split("\n")
+
+            assert (
+                len(raw_sentences) == len(ascii_sentences) == len(tok_sentences)
+            ), "Sizes do not match: %d vs %d vs %d for Raw Sentences vs Ascii Sentences vs Tokenized Sentences." % (
+                len(raw_sentences),
+                len(ascii_sentences),
+                len(tok_sentences),
+            )
+
+            annotation_pairs = []
+            for annot in annot_sentences:
+                line = annot.split(" ")
+                if len(line) > 1:
+                    annotation_pairs.append((int(line[0]), int(line[1])))
+
+            annotations = dict()
+            for row in range(last_id, last_id + len(raw_sentences)):
+                annotations[row] = set()
+
+            for (a, b) in annotation_pairs:
+                # required for dummy data creation
+                if last_id + a not in annotations:
+                    annotations[last_id + a] = set()
+                if last_id + b not in annotations:
+                    annotations[last_id + b] = set()
```

Comment:
are we sure there's no collision here ?

Reply:
I am not sure I understand fully.

The issue for the dummy data was that the number of lines being picked up was not complete, and hence the annotation lines referenced annotation sets that were not created. These lines create sets for these lines, however these sets are not yielded since the for loop with the yield does not reach these values.